mava/components/jax/training/losses.py

```diff
@@ -118,17 +120,51 @@ def policy_loss_grad_fn(
                 # the case of non-shared weights.
                 def policy_loss_fn(
                     policy_params: Any,
+                    policy_states: Any,
                     observations: Any,
                     actions: jnp.ndarray,
                     behaviour_log_probs: jnp.ndarray,
                     advantages: jnp.ndarray,
                 ) -> Tuple[jnp.ndarray, Dict[str, jnp.ndarray]]:
                     """Inner policy loss function: see outer function for parameters."""
-                    distribution_params = network.policy_network.apply(
-                        policy_params, observations
-                    )
+
+                    # TODO (dries): Can we implement something more general here? Like a function call?
+                    if policy_states:
```

Comment:
if statements are not ideal on policy_states and also in networks. Perhaps we get this PR in and later think about refactoring as per the TODO

Reply:
For the executor, you created new components. Did you prefer if statements here as we want to have all losses in this file? (same for networks)

Reply:
I will give this some thought b t your reasoning is fair